nifi-nar-bundles/nifi-media-bundle/nifi-media-processors/src/main/java/org/apache/nifi/processors/document/ExtractDocumentText.java

```diff
@@ -0,0 +1,142 @@
+    private Set<Relationship> relationships = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(REL_SUCCESS, REL_FAILURE)));
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return this.relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return descriptors;
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        return;
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            flowFile = session.create();
+        }
+
+        final int maxTextLength = context.getProperty(MAX_TEXT_LENGTH).evaluateAttributeExpressions(flowFile).asInteger();
+        final String filename = flowFile.getAttribute("filename");
+
+        try {
+            final AtomicReference<String> type = new AtomicReference<>();
+            final AtomicReference<Boolean> wasError = new AtomicReference<>(false);
+
+            flowFile = session.write(flowFile, (inputStream, outputStream) -> {
+                if (inputStream != null) {
+                    BufferedInputStream buffStream = new BufferedInputStream(inputStream);
+                    Tika tika = new Tika();
+                    String text = "";
+                    try {
+                        type.set(tika.detect(buffStream, filename));
+                        tika.setMaxStringLength(maxTextLength);
+                        text = tika.parseToString(buffStream);
+
+                    } catch (TikaException e) {
+                        getLogger().error("Apache Tika failed to parse input " + e.getLocalizedMessage());
+                        wasError.set(true);
+                    }
+
+                    outputStream.write(text.getBytes());
+                    buffStream.close();
+                } else {
+                    getLogger().error("Input file was null");
+                    wasError.set(true);
+                }
+            });
+
+            if (wasError.get()) {
+                session.transfer(flowFile, REL_FAILURE);
+            } else {
+
+                Map<String, String> mimeAttrs = new HashMap<>();
+                mimeAttrs.put("mime.type", TEXT_PLAIN);
+                mimeAttrs.put("orig.mime.type", type.get());
+
+                flowFile = session.putAllAttributes(flowFile, mimeAttrs);
+                session.transfer(flowFile, REL_SUCCESS);
+            }
+        } catch (final Throwable t) {
+            getLogger().error("Unable to process ExtractTextProcessor file " + t.getLocalizedMessage());
+            getLogger().error("{} failed to process due to {}; rolling back session", new Object[]{this, t});
+            // not sure about this one
+            session.transfer(flowFile, REL_FAILURE);
+            throw t;
```

Comment:
The exception should not be thrown if the FlowFile is being routed to a relationship.